lib/reverse_sentence.rb

```diff
@@ -1,6 +1,19 @@
 # A method to reverse the words in a sentence, in place.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n)
+# Space complexity: O(1)
 def reverse_sentence(my_sentence)
```

Comment:
This works, but since you're creating a new array `words_and_space` you have a space complexity of O(n) and because you `prepend` with each iteration (prepend has time complexity of O(n) by itself), your time complexity is O(n^2)... Not good.

I strongly encourage you to think about how you could do this in-place, and if needed ask about it in CS Fundamentals or in office hours.